File: LeetCode/py1.py
# 题目描述：给定一个整数数组 nums 和一个目标值 target，请你在该数组中找出和为目标值的那 两个 整数，并返回他们的数组下标。
# 思路：暴力枚举，两重循环判断和是否满足条件
from typing import List


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        for i in range(len(nums)-1):
            for j in range(i+1, len(nums)):
                if nums[i]+nums[j] == target:
                    return [i,j]
        return [0,0]


if __name__ == '__main__':
    nums = [2, 7, 11, 15]
    target = 9
    ans = Solution().twoSum(nums, target)
    print(ans)

File: py1.cpp
#include<vector>
#include<iostream>
using namespace std;
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> a(2);
        for(int i=0;i<nums.size()-1;i++)
            for (int j=i+1;j<nums.size();j++)
                if (nums[i]+nums[j] == target){
                    a[0] = i;
                    a[1] = j;
                    return a;
                }
             return a;       
    }
};
int main(){
    Solution solution = Solution();
    vector<int> nums = {2, 7, 11, 15};
    int target = 9;
    vector<int> result = (solution.twoSum(nums, target));
    for (int i=0;i<result.size();i++)
        cout<<result[i]<<" ";

    return 0;

}
